File: 0001_0100/0009/0009.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s = [int(x) for x in f.read().split()[1:]]
    t = sum((x for x in s if x > 0))
    smin, smax = s.index(min(s)), s.index(max(s))
    m = 1
    for x in s[min(smin, smax) + 1 : max(smin, smax)]:
        m *= x
    q.write('{} {}'.format(t, m))

File: 0001_0100/0009/0009.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var s, mini, maxi, n int
	fmt.Fscan(f, &n)
	p := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscan(f, &p[i])
		if p[i] > 0 {
			s += p[i]
		}
		if p[mini] > p[i] {
			mini = i
		} else if p[maxi] < p[i] {
			maxi = i
		}
	}
	m := 1
	if mini > maxi {
		mini, maxi = maxi, mini
	}
	for i := mini + 1; i < maxi; i++ {
		m *= p[i]
	}
	fmt.Fprint(q, s, m)
}
